Sources/DVR/SessionDataTask.swift

```diff
@@ -42,42 +42,50 @@ final class SessionDataTask: URLSessionDataTask {
     override func resume() {
 
         // apply request transformations, which could impact matching the interaction
-        var filteredRequest = request
-        session.filters.forEach { filter in
-            filteredRequest = filter.filter(request: filteredRequest)
+        var filteredRequest: URLRequest? = request
+        for filter in session.filters {
+            if let req = filteredRequest {
+                filteredRequest = filter.filter(request: req)
+            }
+            else {
+                break
+            }
         }
-
+        
+        
         if session.recordMode != .all {
             let cassette = session.cassette
-
             // Find interaction
-            if let interaction = session.cassette?.interactionForRequest(filteredRequest, headersToCheck: headersToCheck) {
-                self.interaction = interaction
-                // Forward completion
-                if let completion = completion {
-                    queue.async {
-                        completion(interaction.responseData, interaction.response, nil)
+            if let filteredRequest = filteredRequest {
+                if let interaction = session.cassette?.interactionForRequest(filteredRequest, headersToCheck: headersToCheck) {
+                    self.interaction = interaction
+                    // Forward completion
+                    if let completion = completion {
+                        queue.async {
+                            completion(interaction.responseData, interaction.response, nil)
+                        }
                     }
+                    session.finishTask(self, interaction: interaction, playback: true)
+                    return
                 }
```

Comment:
Here we could make `interactionForRequest` accept an optional request and then `return nil` if the request was `nil`. This would avoid the double unwrapping and the need to have some kind of `else` block for the initial unwrapping. Alternatively, you should be able to combine these:
```swift
if let filteredRequest = filteredRequest, let interaction = .... {
}
```